kuma/users/models.py

```diff
@@ -152,6 +152,10 @@ class User(AbstractUser):
         blank=True,
         validators=[WEBSITE_VALIDATORS['github']],
     )
+    is_github_url_public = models.BooleanField(
+        _(u'Public Github URL'),
+        default=False,
```

Comment:
In my opinion, an opt-in is the right way to do it because Mozilla is privacy focused and displaying the Github profile is a matter of privacy.